sale_commission_product_criteria/models/commission.py

```diff
@@ -0,0 +1,270 @@
+                    "Product Price"
+                )
+                if item.currency_id.position == "after":
+                    item.commission_value = "%s %s" % (
+                        float_repr(
+                            item.fixed_amount,
+                            decimal_places,
+                        ),
+                        item.currency_id.symbol or "",
+                    )
+                else:
+                    item.commission_value = "%s %s" % (
+                        item.currency_id.symbol or "",
+                        float_repr(
+                            item.fixed_amount,
+                            decimal_places,
+                        ),
+                    )
+            elif item.commission_type == "percentage":
+                item.commission_value = str(item.percent_amount) + " %"
+
+    @api.constrains("product_id", "product_tmpl_id", "categ_id")
+    def _check_product_consistency(self):
+        for item in self:
+            if item.applied_on == "2_product_category" and not item.categ_id:
+                raise ValidationError(
+                    _(
+                        "Please specify the category for which this rule should "
+                        "be applied"
+                    )
+                )
+            elif item.applied_on == "1_product" and not item.product_tmpl_id:
+                raise ValidationError(
+                    _(
+                        "Please specify the product for which this rule should "
+                        "be applied"
+                    )
+                )
+            elif item.applied_on == "0_product_variant" and not item.product_id:
+                raise ValidationError(
+                    _(
+                        "Please specify the product variant for "
+                        "which this rule should be applied"
+                    )
+                )
+
+    @api.onchange("product_id")
+    def _onchange_product_id(self):
+        if any(item.applied_on != "0_product_variant" for item in self):
+            raise ValidationError(
+                _("No es pot seleccionar un variant quan l'ítem s'aplica a '%s'.")
+                % dict(self._fields["applied_on"].selection)[self.applied_on]
+            )
+
+        for item in self:
+            item.product_tmpl_id = item.product_id.product_tmpl_id
+
+    @api.onchange("product_tmpl_id")
+    def _onchange_product_tmpl_id(self):
+        if any(item.applied_on != "1_product" for item in self):
+            raise ValidationError(
+                _("No es pot seleccionar un template quan l'ítem s'aplica a '%s'.")
+                % dict(self._fields["applied_on"].selection)[self.applied_on]
+            )
+
+        for item in self:
+            if (
+                item.product_id
+                and item.product_id.product_tmpl_id != item.product_tmpl_id
+            ):
+                item.product_id = False
```

Comment:
```suggestion
    @api.onchange("product_id")
    def _onchange_product_id(self):
        has_product_id = self.filtered("product_id")
        for item in has_product_id:
            item.product_tmpl_id = item.product_id.product_tmpl_id
        if self.env.context.get("default_applied_on", False) == "1_product":
            # If a product variant is specified, apply on variants instead
            # Reset if product variant is removed
            has_product_id.update({"applied_on": "0_product_variant"})
            (self - has_product_id).update({"applied_on": "1_product"})

    @api.onchange("product_tmpl_id")
    def _onchange_product_tmpl_id(self):
        has_tmpl_id = self.filtered("product_tmpl_id")
        for item in has_tmpl_id:
            if (
                item.product_id
                and item.product_id.product_tmpl_id != item.product_tmpl_id
            ):
                item.product_id = None
```
These changes make it impossible to add products to a commission item once it has been created. Please revert it to how it was before, and if necessary (though I don't think it is), adjust it so that it continues to work.